### ISSUE TITLE
Importing API for the first time is not working when we use --update flag 

### ISSUE THREAD

**Issue by Senthuran100** [wso2/product-apim-tooling#343]
**Description:**
Import an API  called ORDS-CLI then import the API with the name ORDS with update command.
`./apimcli import-api -f ORDS_1.0.0.zip -e productionnew -k --update --verbose
`
When we follow the above step we get an error in APIM CLI and we weren't able to update the API successfully.

**Suggested Labels:**
APIMCLI 2.0.7

**Affected Product Version:**
APIMCLI 2.0.7 , APIM 2.6.0

**Steps to reproduce:**
- Import an API  called ORDS-CLI
- Import the API with the name ORDS with the update command
- Able to reproduce the issue.

**Comment by npamudika** [comment]
Narrowed down the issue and this is a bug in Publisher REST API layer with the API search, as it returns **ORDS-CLI** API, even the search happens with the API name **ORDS**.

**Comment by npamudika** [comment]
Debugged further and found out that in the Publisher REST API, it does a **wildcard** search and thus this issue occurs when calling the REST API as below.

```
https://localhost:9443/api/am/publisher/v1/apis?query=name%3AORDS+version%3A1.0+provider%3Aadmin
```

**Comment by msm1992** [comment]
REST API call for URL: https://localhost:9443/api/am/publisher/v1/apis?query=name%3AORDS+version%3A1.0+provider%3Aadmin does an wildcard match for the provided query strings. So inside the core this will be translated into 
```name=*ORDS*&version=*1.0*&provider=*admin*``` 
That is not a bug but the way the /apis endpoint is designed to do the search operation.

If we need an exact match for the given name, version and provider trio then I suppose we have to adopt one of the following options.

**Option 1**

Modify existing /apis endpoint to perform searches for exact matches on demand. For this we can introduce a new query param something like ```exact```. So when this param is set to true then /apis endpoint will perform an exact match otherwise by default exact will be false and the /apis will perform the usual wildcard match.

ex : 
```https://localhost:9443/api/am/publisher/v1/apis?query=name%3AORDS+version%3A1.0+provider%3Aadmin&exact=true```
This will return ORDS API only

```https://localhost:9443/api/am/publisher/v1/apis?query=name%3AORDS+version%3A1.0+provider%3Aadmin``` OR
```https://localhost:9443/api/am/publisher/v1/apis?query=name%3AORDS+version%3A1.0+provider%3Aadmin&exact=false```
These will return both ORDS and ORDS-CLI APIs as search results.

**Option 2**

After CLI tool receives the search results for the /apis query, we can check the result set to see whether an exact match for the name,version and admin exists before rushing to decide that the API exists.

**Comment by npamudika** [comment]
Thanks @msm1992 for the explanation. I am +1 for the 1st option as it returns the expected result using a single query as we do now. And this won't harm the existing logic too.

If we use the 2nd option, it will introduce another overhead to do an exact match at the APICTL side.

**Comment by wasuradananjith** [comment]
If we pass the name of the API inside quotation marks, this will do an exact search instead of a wildcard search. 
For example, if we do the search with "ORDS" instead of just using ORDS, the exact API with the name ORDS will return.
I think this is the simplest solution as it seems.

**Comment by npamudika** [comment]
Great @wasuradananjith. Let's go with the mentioned fix adding quotation marks.

**Comment by npamudika** [comment]
The fix will be available in the APICTL 3.2.0 release.

**Comment by npamudika** [comment]
Fixed in APIMCLI 2.0.8 version

**Comment by wasuradananjith** [comment]
Fixed in APICTL 3.0.2 version (https://github.com/wso2/product-apim-tooling/releases/tag/v3.0.2)
